Src/Shared/Network/DatabaseConnection.cpp

```diff
@@ -13,16 +13,50 @@ CDatabaseConnection::CDatabaseConnection(
 	if (!mysql_init(&_mySQL))
 		throw CDatabaseException(SRC_POS, StrFormat("MySQL struct could not be initialized. ({0})", Error()));
 
+	_isInitialized = true;
+
 	connect();
 
 	_pActive = CActive::Create();
 }
 
+CDatabaseConnection::CDatabaseConnection(CDatabaseConnection&& other)
+{
+	*this = std::move(other);
+}
+
+CDatabaseConnection& CDatabaseConnection::operator=(CDatabaseConnection&& other)
+{
+	_port = other._port;
+	_mySQL = other._mySQL;
+
+	if (!other._pActive || other._pActive->IsBusy())
+	{
+		// Deconstruct the previous connection's active object
+		// to force background tasks to finish.
+		other._pActive = nullptr;
+		_pActive = CActive::Create();
+	}
+	else
+		_pActive = std::move(other._pActive);
```

Comment:
isn't there a race condition possible here? just asking

Reply:
It's not safe to use `CDatabaseConnenction` from other threads while moving it, but the termination of running background queries doesn't have a race condition as far as I can tell.